presto-main/src/main/java/com/facebook/presto/statusdetector/NodeStatusDetector.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.statusdetector;
+
+import java.util.List;
+
+public interface NodeStatusDetector
+{
+    List<String> getDisabled();
```

Comment:
How about `Set<ServiceDescriptor>` instead of `List<String>` since I think there cannot be duplicates and `ServiceDescriptor` captures the idea of node better than `String`

Reply:
change the interface as per Mayank's suggestion and so no longer need getDisabled method.